### PR TITLE
Refactor named maps controller

### PR DESCRIPTION
**Pull request by dgaubert**
In order to make middlewares easily interchangeable, I extracted them from controller's context.

### CODE REVIEW

File: lib/cartodb/controllers/named_maps.js [hunk 1]
```diff
@@ -50,19 +52,21 @@ NamedMapsController.prototype.register = function(app) {
...
-        this.getTile(tileOptions),
-        this.setSurrogateKey(),
-        this.setCacheChannelHeader(),
-        this.setLastModifiedHeader(),
-        this.setCacheControlHeader(),
-        this.setContentTypeHeader(),
-        this.respond(),
+        getNamedMapProvider(tileOptions),
+        getAffectedTables(),
+        getTile(tileOptions),
```
Comment (oleurud) [inline lib/cartodb/controllers/named_maps.js]: maybe it is more clear to send the parameters needed:
```
getTile(this.tileBackend, 'NAMED_MAP_TILE')
```

File: lib/cartodb/controllers/named_maps.js [hunk 2]
```diff
@@ -73,23 +77,23 @@ NamedMapsController.prototype.register = function(app) {
...
-        this.setCacheChannelHeader(),
-        this.setLastModifiedHeader(),
-        this.setCacheControlHeader(),
-        this.setContentTypeHeader(),
-        this.respond()
+        getNamedMapProvider(staticOptions),
+        getAffectedTables(),
+        getTemplate(staticOptions),
+        prepareLayerFilterFromPreviewLayers(staticOptions),
+        getStaticImageOptions({ tablesExtentApi: this.tablesExtentApi }),
```
Comment (oleurud) [inline lib/cartodb/controllers/named_maps.js]: I am losing something, but why send objects and then destructuring?

Comment (dgaubert) [reply lib/cartodb/controllers/named_maps.js]: There are a couple of reasons:
  - make optional parameters much less PITA.
  - unorder params identified by name.
  - avoid long params list
  -  you can still define default values
  - ...

File: lib/cartodb/controllers/named_maps.js (hunk 1 above)
Comment (dgaubert) [reply lib/cartodb/controllers/named_maps.js]: sure

File: lib/cartodb/controllers/named_maps.js (hunk 2 above)
Comment (oleurud) [reply lib/cartodb/controllers/named_maps.js]: Agree, but even with only one?

Comment (oleurud) [reply lib/cartodb/controllers/named_maps.js]: And it does not seem bad to me, it just makes me a little weird

Comment (enekid) [reply lib/cartodb/controllers/named_maps.js]: Look at this as a way to pass `named` parameters.

Comment (dgaubert) [reply lib/cartodb/controllers/named_maps.js]: > Agree, but even with only one?

To keep consistency and to be ready to add new params

**Pull request merged**